src/main/java/org/apache/mesos/scheduler/plan/api/PlansResource.java

```diff
@@ -0,0 +1,133 @@
+package org.apache.mesos.scheduler.plan.api;
+
+import org.apache.mesos.scheduler.plan.PlanManager;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * API for management of Plan(s).
+ */
+@Path("/v1/plans")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class PlansResource {
+    private static final Response PLAN_NOT_FOUND_RESPONSE = Response.status(Response.Status.NOT_FOUND)
+            .entity("Plan not found")
+            .build();
+
+    private Map<String, PlanManager> planManagers = new HashMap<>();
+
+    public PlansResource(final Map<String, PlanManager> planManagers) {
+        this.planManagers.putAll(planManagers);
+    }
+
+    /**
+     * Returns the status of the currently active Phase/Block.
+     */
+    @GET
+    @Path("/{planName}/status")
+    public Response getStatus(@PathParam("planName") String planName) {
+        if (planManagers.containsKey(planName)) {
```

Comment:
Here and below, could just do 'planManagers.get(planName)' up-front and then check 'manager != null' to determine if planName is valid.